algorithms.py

```diff
@@ -145,7 +178,100 @@ def configure(self, data, args):
         params = super(XgbGPUHistAlgorithm, self).configure(data, args)
         params.update({"tree_method": "gpu_hist", "gpu_id": 0})
         return params
+        
+class SkRandomForestAlgorithm(Algorithm):
+    def configure(self, data, args):
+        params = shared_params.copy()
+        if data.learning_task == LearningTask.REGRESSION:
+            params["objective"] = "regression"
+        elif data.learning_task == LearningTask.CLASSIFICATION:
+            params["objective"] = "binary"
+            params["scale_pos_weight"] = len(data.y_train) / np.count_nonzero(data.y_train)
+        elif data.learning_task == LearningTask.MULTICLASS_CLASSIFICATION:
+            params["objective"] = "multiclass"
+            params["num_class"] = np.max(data.y_test) + 1
+        params.update(args.extra)
+        return params
+
+    def fit(self, data, args):
+        params = self.configure(data, args)
+        if params["objective"] == "regression":
```

Comment:
Instead of using `params["objective"]` here, let's use `data.learning_task` directly to decide between `RandomForestClassifier` and `RandomForestRegressor`.